### PR TITLE
Adding Java examples

### PR DESCRIPTION
**Pull request by sidd0610**

### CODE REVIEW

File: java/src/main/resources/arguments.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#
+# Note: Please ensure to specify a value of `null` to all optional configurations when
+# you do not wish to specify a value for the same. Some optional configurations will be
+# initialised with default values specified below.
+
+# =========================
+# Required Configurations:
+# =========================
+# Pub/Sub API Endpoint
+PUBSUB_HOST: api.stage.pubsub.salesforce.com
```
Comment (jclinn) [inline java/src/main/resources/arguments.yaml:11]: use prod endpoint

File: java/src/main/resources/arguments.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+# For authentication, you can use either username/password or sessionToken/tenantId types.
+# Either one of the combinations is required. Please specify `null` values to the unused type.
+# Your Salesforce Username
+USERNAME: null
+# Your Salesforce Password
+PASSWORD: null
+# Your Salesforce org Tenant ID
+TENANT_ID: CORE/prod/00DS7000000lpYUMAY
+# Your Salesforce Session Token
+ACCESS_TOKEN: "00DS7000000lpYU!AQMAQHN_P0E98M1TXvmYFYo8X4Aeq7UtUN5_4lNNRMC.LchxXpA0L__o0qhDpAw2u4jzu4.WvCyMiD6PRIIqtsqeFa_xxO6A"
```
Comment (jclinn) [inline java/src/main/resources/arguments.yaml:26]: remove access token

File: java/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+       * `REPLAY_PRESET`: Specify the ReplayPreset for subscribe examples.
+2. After setting up the configurations, any example can be executed using the `./run.sh` file available at the parent directory.
+   * Format for running the examples: `./run.sh <package_name>.<example_class_name>`
+   * Example: `./run.sh genericpubsub.PublishStream`
+
+## Implementation
+This repo can be used as a reference point for clients looking to create a Java app to integrate with Pub/Sub API. Note that the project structure and the examples included in this repo are intended for demo purposes only and clients are free to implement their own Java apps in any way they see fit. 
+
+# Limitations
+1. No support for C2C type of authentication. This may be included in future updates.
```
Comment (jclinn) [inline java/README.md:56]: C2C is for internal use only. Please remove.

File: java/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+2. No support for auth token refreshes - At some point the authentication token will expire. At this time, these examples do not handle re-authentication.
```
Comment (jclinn) [inline java/README.md:57]: Session tokens do not need auth refresh. Clients can retrieve a new session token if the existing session token is revoked by admin or through the connected app.

**Comment by brendanjlynch4014** [comment]
Hi  sidd0610  ,   I tried to download and compile this branch 'siddartha/java-examples' but it failed to find java package 'com.salesforce.eventbus.protobuf'      Where can I pull down these dependent classes?

**Comment by sidd0610** [comment]
> Hi sidd0610 , I tried to download and compile this branch 'siddartha/java-examples' but it failed to find java package 'com.salesforce.eventbus.protobuf' Where can I pull down these dependent classes?

@brendanjlynch4014 You should be able to build it using `mvn clean install` after going into the java directory. You don't need to pull any additional classes. Let me know if you're still having issues!

**Review by jclinn — APPROVED** [review]

**Comment by meruzhan79** [comment]
Hi Siddartha, 
Please let me know what the logic for the retry feature in the OnError method (in the com.optum.pubsub.Subscribe class). 

Thanks,
Meruzhan

**Comment by charu-jain89** [comment]
Hi @sidd0610 
I have 1 doubt wrt ProcessChangeEventHeader class.
Lets say my application or ProcessChangeEventHeader is down, which is listening to changes in Opportunity object, and now I have made changes to Opportunity object or created a new Opportunity.
Again if I re-run the app or ProcessChangeEventHeader  class. Now its not polling the changes to Opportunity.
In normal Pub/Sub model, it does the poll but its not the case here.
Kindly help here.

**Pull request merged**